File: app/src/main/java/com/droidhats/campuscompass/helpers/Subject.java
package com.droidhats.campuscompass.helpers;

/**
 *Subject interface used implement attach, detach methods for observers and notifying observers
 *
 * Please see the other class {@link com.droidhats.campuscompass.helpers.Observer for reference.
 */
public interface Subject {

    void attach(Observer observer);

    void detach(Observer observer);

    void notifyObservers();
}

File: app/src/main/java/com/droidhats/campuscompass/helpers/Subject.kt
package com.droidhats.campuscompass.helpers

/**
 * Subject interface used implement attach, detach methods for observers and notifying observers
 *
 * Please see the other class {@link com.droidhats.campuscompass.helpers.Observer for reference.
 */
interface Subject {

    fun attach(observer: Observer?)

    fun detach(observer: Observer?)

    fun notifyObservers()
}